source/lib.ts

```diff
@@ -0,0 +1,323 @@
+			return new Promise((resolve, reject) => {
+				legacyStore.incr(
+					key,
+					(
+						error: Error | undefined,
+						totalHits: number,
+						resetTime: Date | undefined,
+					) => {
+						if (error) reject(error)
+						resolve({ totalHits, resetTime })
+					},
+				)
+			})
+		}
+
+		async decrement(key: string): Promise<void> {
+			return Promise.resolve(legacyStore.decrement(key))
+		}
+
+		async resetKey(key: string): Promise<void> {
+			return Promise.resolve(legacyStore.resetKey(key))
+		}
+
+		async resetAll(): Promise<void> {
+			if (typeof legacyStore.resetAll === 'function')
+				return Promise.resolve(legacyStore.resetAll())
+		}
+	}
+
+	return new PromisifiedStore()
+}
+
+/**
+ * Adds the defaults for options the user has not specified.
+ *
+ * @param options {Options} - The options the user specifies
+ *
+ * @returns {Options} - A complete configuration object
+ */
+const parseOptions = (passedOptions: Partial<Options>): Options => {
+	// Now add the defaults for the other options
+	const options: Options = {
+		windowMs: 60 * 1000,
+		store: new MemoryStore(passedOptions.windowMs ?? 60 * 1000),
+		max: 5,
+		message: 'Too many requests, please try again later.',
+		statusCode: 429,
+		legacyHeaders: true,
+		standardHeaders: false,
+		requestPropertyName: 'rateLimit',
+		skipFailedRequests: false,
+		skipSuccessfulRequests: false,
+		requestWasSuccessful: (_request, response) => response.statusCode < 400,
+		keyGenerator: (request) => {
+			if (!request.ip) {
+				console.error(
+					'WARN | `express-rate-limit` | `request.ip` is undefined. You can avoid this by providing a custom `keyGenerator` function, but it may be indicative of a larger issue.',
+				)
+			}
+
+			return request.ip
+		},
+		skip: () => false,
+		handler: (_request, response, _optionsUsed: Options) =>
+			response.status(options.statusCode).send(options.message),
+		onLimitReached: () => {},
+		...passedOptions,
+	}
+
+	// Ensure that the store passed implements the either the `Store` or `LegacyStore`
+	// interface
+	if (
+		typeof options.store.increment !== 'function' ||
+		typeof options.store.resetKey !== 'function' ||
+		(options.skipFailedRequests &&
+			typeof options.store.decrement !== 'function')
+	) {
+		throw new Error(
+			'An invalid store was passed. Please ensure that the store is a class that implements the `Store` interface.',
+		)
+	} else {
+		// Promisify the store, if it is not already
+		options.store = promisifyStore(options.store)
+	}
+
+	// Throw an error if any deprecated options are passed
+	const deprecatedOptions = [
+		'global',
+		'delayMs',
+		'delayAfter',
+		'headers',
+		'draft_polli_ratelimit_headers',
+	] as Array<keyof Options>
+	for (const option of deprecatedOptions) {
+		if (typeof passedOptions[option] !== 'undefined') {
+			throw new TypeError(
+				`The \`${option}\` option is deprecated/renamed and will likely be removed from the \`express-rate-limit\` package in the future.`,
+			)
+		}
+	}
```

Comment:
All of this can go - the global and delay options haven't been supported in years, and the above change should handle the headers options.
```suggestion

```

Reply:
I could remove the `global`, `delayMs` and `delayAfter` options and add a custom deprecation message for the header options.

Reply:
yep, we write to README, migration guide like breaking changes